dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java

```diff
@@ -51,91 +62,115 @@ public UserNotificationService(UserNotificationRepository userNotificationReposi
+//
+//        UserNotification userNotification = createUserNotification(dataverseUser, sendDate, type, dvObjectId, comment);
+//
+//        userNotificationRepository.saveAndFlush(userNotification);
+//
+//        executorService.submit(() -> sendEmail(userNotification.getId(), notificationObjectType));
+//    }
+
+    public void sendNotificationWithEmail(AuthenticatedUser dataverseUser, Timestamp sendDate, String type, Long dvObjectId,
+                                          NotificationObjectType notificationObjectType, Map<String, String> parameters) {
+        UserNotification userNotification = createUserNotification(dataverseUser, sendDate, type, dvObjectId, parameters);
         userNotificationRepository.saveAndFlush(userNotification);
-
         executorService.submit(() -> sendEmail(userNotification.getId(), notificationObjectType));
     }
 
-    /**
-     * Saves notification to database, then sends email asynchronously.
-     *
-     * @param dvObjectId
-     * @param notificationObjectType - type has to match correct #{@link NotificationType}
-     */
-    public void sendNotificationWithEmail(AuthenticatedUser dataverseUser,
-                                          Timestamp sendDate,
-                                          String type,
-                                          Long dvObjectId,
-                                          NotificationObjectType notificationObjectType,
-                                          AuthenticatedUser requestor,
-                                          String comment) {
-
-        UserNotification userNotification = createUserNotification(dataverseUser, sendDate, type, dvObjectId, requestor, comment);
-
-        userNotificationRepository.saveAndFlush(userNotification);
 
-        executorService.submit(() -> sendEmail(userNotification.getId(), notificationObjectType));
+    public Map<String, String> getParameters(UserNotification notification) {
+        String json = notification.getParameters();
+        return StringUtils.isNotBlank(json) ? getParametersMap(json) : new HashMap<>();
     }
 
-    /**
-     * Saves notification to database, then sends email asynchronously.
-     *
-     * @param notificationObjectType - type has to match correct #{@link NotificationType}
-     * @param comment                - custom user message added to notification on '{@link NotificationType#RETURNEDDS}
-     */
-    public void sendNotificationWithEmail(AuthenticatedUser dataverseUser,
-                                          Timestamp sendDate,
-                                          String type,
-                                          Long dvObjectId,
-                                          NotificationObjectType notificationObjectType,
-                                          String comment) {
+    public void setParameters(UserNotification notification, Map<String, String> parameters) {
+        notification.setParameters(parametersToString(parameters));
+    }
```

Comment:
IMO these two methods do not quite belong to this service. IMO they pollute public api of the service. I would try to extract it to some either some other service or make a util class. But I'm not insisting on it. We can probably live with this